backend/chunker/chunker.go

```diff
@@ -709,8 +771,14 @@ func (f *Fs) processEntries(ctx context.Context, origEntries fs.DirEntries, dirP
 			}
 			// this is some kind of chunk
 			// metobject should have been created above if present
-			isSpecial := xactID != "" || ctrlType != ""
 			mainObject := byRemote[mainRemote]
+			if f.useNoRename {
+				currentXactID, err = mainObject.readXactID(ctx)
+				if err != nil {
```

Comment:
This will re-read metadata object for every item in the directory!
It will make integration tests pass, but the price is too high.

Reply:
Since `readXactID` caches the `xactID` after it is read (and immediately returns it if it was previously cached), it should still be reading only once per `mainObject`. I believe that is what we want no?

Reply:
Agreed. I was misguided by the use of single variable `currentXactID` in `processEntries`.
I will replace it by a hash now, so it looks cleaner.

Reply:
Here is the commit: https://github.com/rclone/rclone/pull/4675/commits/1d219095fb5fd9903504f78b6a80e79173d3bb8b
The commit name is outdated after your note.